### PR TITLE
fix(ml): fix memory leak in PyTorch tensor allocation during PIRL inference

### PR DESCRIPTION
**Pull request by tmdeveloper007**
## Spec 111

**Problem:** Tensors retain autograd computational graph during evaluation mode.

**Solution:** Wrap inference calls inside `with torch.no_grad():` block.

**Verification:** ``cd backend/ml/pirl && python3 test_pirl.py``

Closes spec #111

### CODE REVIEW

**Comment by KanishJebaMathewM** [comment]
level:advanced

**Comment by KanishJebaMathewM** [comment]
level:advanced

**Comment by KanishJebaMathewM** [comment]
level:advanced

**Comment by KanishJebaMathewM** [comment]
level:advanced

**Comment by KanishJebaMathewM** [comment]
level:advanced

**Comment by KanishJebaMathewM** [comment]
level:advanced
